build.gradle

```diff
@@ -0,0 +1,32 @@
+allprojects {
+    group 'me.coley'
+    version '3.0.0-SNAPSHOT'
+}
+
+subprojects {
+    apply plugin: 'java'
+
+    sourceCompatibility = 1.8
+    targetCompatibility = 1.8
+    tasks.withType(JavaCompile) {
+        options.encoding = 'UTF-8'
+        options.incremental = true
+    }
+
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        maven { url 'http://files.minecraftforge.net/maven' }
+        maven { url 'https://jitpack.io' }
+    }
+
+    project.ext {
+        asm = 'org.ow2.asm:asm:9.0'
+        asm_commons = 'org.ow2.asm:asm-commons:9.0'
+        asm_tree = 'org.ow2.asm:asm-tee:9.0'
+        asm_util = 'org.ow2.asm:asm-util:9.0'
+        asm_analysis = 'org.ow2.asm:asm-analysis:9.0'
```

Comment:
For re-used variables like ASM's version we could use variables. Its nicer only having to update one value